walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,52 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Long duration = request.getDuration();
+    Iterator<Event> iterator = events.iterator();
+    int start = 0;
+    Collection<TimeRange> timeBlock = new ArrayList();
+    if(duration>TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList();
+    }
+    if(events.size()==0 ){
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    /**returns all time intervals between events*/
+    while (iterator.hasNext()) {
+        Event e = iterator.next();
+        int min = 0;
+        while (min<=TimeRange.END_OF_DAY){
+            if(min==TimeRange.END_OF_DAY){
+                timeBlock.add(TimeRange.fromStartEnd(start,min,true));
+                return timeBlock;
+            }
+            else if (e.getWhen().overlaps(TimeRange.fromStartEnd(start,min,true))){
+                if(!Collections.disjoint(e.getAttendees(), request.getAttendees())){
+                        timeBlock.add(TimeRange.fromStartEnd(start,min,false));
+                        min = e.getWhen().end();
+                        start = e.getWhen().end();
+                }
+            }
+            min++;
+        }
+    }
+    /**deletes time intervals shorter than duration*/
+    Iterator<TimeRange> times = timeBlock.iterator();
+    while(times.hasNext()){
+        if(times.next().duration()<duration){
```

Comment:
Spacing looks off here as well.

https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style
In general, refer to style guide if unsure about spacing